### PR TITLE
feat(button): add isExpressive prop

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#3982

### Description

add isExpressive prop

### CODE REVIEW

File: src/components/button/button.ts [hunk 1]
```diff
@@ -156,8 +163,11 @@ class BXButton extends FocusMixin(LitElement) {
       [`${prefix}--btn--${kind}`]: kind,
       [`${prefix}--btn--disabled`]: disabled,
       [`${prefix}--btn--icon-only`]: hasIcon && !hasMainContent,
-      [`${prefix}--btn--${size}`]: size,
+      [`${prefix}--btn--sm`]: size === 'sm' && !isExpressive,
+      [`${prefix}--btn--xl`]: size === 'xl',
+      [`${prefix}--btn--field`]: size === 'field' && !isExpressive,
       [`${prefix}-ce--btn--has-icon`]: hasIcon,
+      [`${prefix}--btn--expressive`]: isExpressive,
```
Comment (annawen1) [inline src/components/button/button.ts:170]: Do we use this class name to style at all? Seems like we only use the attribute to style

**Review by IgnacioBecerra — APPROVED** [review]
LGTM, thanks @ariellalgilmore!!

**Comment by jeffchew** [comment]
LGTM! One thing though, noticed that the `isExpressive` knob isn't working in the react wrapper deploy preview. I didn't check the other expressive components though.

File: src/components/button/button.ts (hunk 1 above)
Comment (ariellalgilmore) [reply src/components/button/button.ts:170]: Screen Shot 2021-06-15 at 10 04 45 AM

yeah the class name is being styled from carbon core

**Comment by ariellalgilmore** [comment]
> LGTM! One thing though, noticed that the `isExpressive` knob isn't working in the react wrapper deploy preview. I didn't check the other expressive components though.

oh didn't touch the other stories. Will add! thanks :)

File: src/components/button/button.ts (hunk 1 above)
Comment (annawen1) [reply src/components/button/button.ts:170]: ohh ok gotcha, cool!

**Review by annawen1 — APPROVED** [review]
LGTM!

**Pull request merged**